pkg/rancher-components/src/components/Button/Button.vue

```diff
@@ -0,0 +1,50 @@
+<script lang="ts">
+import { defineComponent, computed } from 'vue';
+
+type Props = {
+  primary: boolean;
+  secondary: boolean;
+  tertiary: boolean;
+};
+
+export default defineComponent({
+  props: {
```

Comment:
Would not be better a type/eum?

Reply:
It looks like Vue 2 does expose `script setup`, so we can make use of `defineProps` to produce something like this

```
type Props = {
  primary?: boolean;
  secondary?: boolean;
  tertiary?: boolean;
}

const props = defineProps<Props>();
```